kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -131,4 +131,16 @@ public static <T extends HasMetadata> Optional<T> findOnlyResourceNamed(@NonNull
         }
         return list.isEmpty() ? Optional.empty() : Optional.of(list.get(0));
     }
+
+    /**
+     * Index elements of stream by name
+     *
+     * @param <T> resource type
+     * @param stream stream to index
+     * @return a Map where every element of the input stream has an entry with the metadata.name as key and the element as value
+     * @throws IllegalStateException if there are multiple elements in the stream with the same metadata.name
+     */
+    public static <T extends HasMetadata> Map<String, T> indexByName(Stream<T> stream) {
```

Comment:
It _might_ be better to have a `toMapByName` method, which returns the `Collector`. That way it fits more naturally into stream pipelines at call sites. Or have this method take a `Collection`.

Reply:
yeah agree, it's a weird mix of Stream in and Collection out. I'll go for returning a `Collector` instead.